spec/controllers/orgs/rosters_controller_spec.rb

```diff
@@ -209,6 +209,104 @@
     end
   end
 
+  describe "GET #import_lms_roster", :vcr do
+    before do
+      sign_in_as(user)
+    end
+
+    context "with lti launch enabled" do
+      before(:each) do
+        GitHubClassroom.flipper[:lti_launch].enable
+      end
+
+      after(:each) do
+        GitHubClassroom.flipper[:lti_launch].disable
+      end
+
+      context "with existing LMS" do
+        let(:lti_configuration) { create(:lti_configuration, organization: organization) }
+
+        context "with context_membership_url" do
+          before(:each) do
+            LtiConfiguration
+              .any_instance
+              .stub(:context_membership_url)
+              .and_return("http://www.example.com")
+          end
+
+          context "fetching roster succeeds" do
+            let(:students) { [] }
+            before(:each) do
+              GitHubClassroom::LTI::MembershipService
+                .any_instance
+                .stub(:students)
+                .and_return(students)
+            end
+
+            it "format.html: succeeds" do
+              get :import_from_lms, params: { id: lti_configuration.organization.slug }
+              expect(response).to have_http_status(:ok)
+              expect(flash[:alert]).to be_nil
+            end
+
+            it "format.js: succeeds" do
+              get :import_from_lms, format: :js, xhr: true, params: { id: lti_configuration.organization.slug }
+              expect(response).to have_http_status(:ok)
+              expect(flash[:alert]).to be_nil
+            end
+          end
+
+          context "fetching roster fails" do
+            before(:each) do
+              GitHubClassroom::LTI::MembershipService
+                .any_instance
+                .stub(:students)
+                .and_raise
+            end
+
+            it "format.html: presents an error message to the user" do
+              get :import_from_lms, params: { id: lti_configuration.organization.slug }
+              expect(flash[:alert]).to be_present
+            end
+
+            it "format.js: presents an error message to the user" do
+              get :import_from_lms, format: :js, xhr: true, params: { id: lti_configuration.organization.slug }
+              expect(response.status).to be(428)
```

Comment:
422 is preferred when missing a required parameter, 428 is for something different.